js/reference.js

```diff
@@ -376,65 +378,101 @@ var ERMrest = (function(module) {
 
                 var uri = this._uri;
 
-                // add sorting
-                // get a list of columns in key
-                // if table has no key, use all columns as key
-                var keycols;
-                if (this._table.keys.length() === 0) {
-                    keycols = this._table.columns.all();
-                } else {
-                    var keys = this._table.keys.all().sort( function(a, b) {
-                        return a.colset.length() - b.colset.length();
-                    });
-                    keycols = keys[0].colset.columns;
+                // if no sorting provided, use schema defined sort if that's present
+                // If neither one present, use shortestkey
+                if (!this._sort || (this._sort.length === 0)) {
+                    if (this._table.annotations.contains(module._annotations.TABLE_DISPLAY) &&
+                        this._table.annotations.get(module._annotations.TABLE_DISPLAY).contains("row_order")) {
```

Comment:
Unlike with the annotation `tag:` URIs  I don't see any value in an enumeration for a constant field-name in the JSON structure itself.  I see that as part of the object model and would have just written code like   `annot.row_order` to access it if I were writing the above...